src/System.CommandLine/Invocation/InvocationExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Reflection;
+
+namespace System.CommandLine.Invocation
+{
+    public static class InvocationExtensions
+    {
+        public static void BindParameter(this ICommandHandler handler, ParameterInfo param, Option option)
```

Comment:
These shouldn't be necessary, correct?

Reply:
I think it is needed, although it might not be in the right place. That extension method drives the following syntax: 

```
            handler.BindParameter(parameter, argument);
            command.Handler = handler;
```

The handler is managed as an ICommandHandler, so putting the method on the interface isn't sensible due to the ParameterInfo/PropertyInfo. Thus, an extension method seemed the right approach.

Reply:
Or were you just expecting to always do it on the argument and this is on the option?

Reply:
I think making `ModelBindingCommandHandler` public makes more sense than extending the interface but throwing if it's the wrong implementation.